### PR TITLE
Require PHP 7.1

### PR DESCRIPTION
**Pull request by lcobucci**
Related to doctrine/doctrine2#6425

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Cache/MongoDBCacheTest.php [hunk 1]
```diff
@@ -6,20 +6,34 @@
...
+
     /**
      * @var MongoCollection
      */
     private $collection;
 
     protected function setUp()
     {
-        $mongo = new MongoClient();
+        if (static::$skip === true) {
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Cache/MongoDBCacheTest.php]: This is nasty, I known... the idea is to speed up the failure.

@Ocramius @alcaeus I'd love if you could provide a better approach.

Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/MongoDBCacheTest.php]: Use a `@requires` annotation at class level?

Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/MongoDBCacheTest.php]: Also: speed is not a problem on this project: if you want to work on that, we have some tests using `sleep()` 😅

Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Cache/MongoDBCacheTest.php]: `sleep()`??? 😭 

F*** that then, the `@requires` will not work because the connection is "lazy" so it fails only when trying to do something with it.

I'll keep the try/catch and remove that nasty static property... thanks

File: composer.json [hunk 2]
```diff
@@ -13,12 +13,16 @@
...
-        "php": "~5.6|~7.0"
+        "php": "~7.1"
     },
     "require-dev": {
+        "alcaeus/mongo-php-adapter": "^1.1",
         "phpunit/phpunit": "^5.7",
         "predis/predis":   "~1.0"
     },
+    "suggest": {
+        "alcaeus/mongo-php-adapter": "Required to use MongoDB driver"
```
Comment (alcaeus) [inline composer.json]: Just cross-referencing #142. Rewriting `MongoDBCache` would be a BC break, so the suggestion is good, but it's not a long-term solution. I'll create a pull request to deprecate the current class, we'll just have to think of some clever naming for the new cache class that uses `ext-mongodb`.

Comment (lcobucci) [inline composer.json]: @alcaeus indeed, I decided to leave it for a different PR to keep things consistent. Thanks for taking it over ;)
